src/clients/SpokePoolClient.ts

```diff
@@ -442,7 +442,7 @@ export class SpokePoolClient {
       currentTime: currentTime.toNumber(), // uint32
       firstDepositId,
       latestBlockNumber,
-      latestDepositId,
+      latestDepositId: Math.max(numberOfDeposits - 1, 0),
```

Comment:
Yeah; technically we'd resolve latestDepositId === 0, when in fact there would be _no_ deposit Ids. This is intuitively a bit awkward. A more correct way of handling it would be to return undefined in this instance, though that puts a burden on every user to check for undefined - basically an edge case that doesn't apply 99.9999% of the time, because we _almost_ never have a SpokePool with no deposits.

I've checked both sdk-v2 and relayer-v2 for side-effects of a SpokePool with no deposits, and I couldn't identify any issues with setting a floor of 0. If we went below 0 then we'd actually [fail to find deposits](https://github.com/across-protocol/relayer-v2/blob/d09654dc5423bb2997c9bb05c95d24e1149aa74a/src/utils/DepositUtils.ts#L133).

Side note: We have implicitly perverted numberOfDeposits through the SpokePool redployment - it was initialised to 1_000_000, so it's not actually representative of how many deposits the SpokePool has had.

Reply:
This is a fair point - I was thinking undefined but wasn't sure if that could also potentially imply that the request failed. This is fine as-is. Thanks for the detailed response.